Integrate strimzi-kafka-oauth 0.6.0 features (#3336)
strimzi/strimzi-kafka-operator: documentation/modules/oauth/proc-oauth-authentication-broker-config.adoc

    validIssuerUri: <__https://<auth-server-address>/auth/realms/external__> <2>
    jwksEndpointUri: <__https://<auth-server-address>/auth/realms/external/protocol/openid-connect/certs__> <3>
    userNameClaim: preferred_username <4>
    maxSecondsWithoutReauthentication: 3600 <5>
    tlsTrustedCertificates: <6>
    - secretName: oauth-server-cert
      certificate: ca.crt
    disableTlsHostnameVerification: true <7>
    jwksExpirySeconds: 360 <8>
    jwksRefreshSeconds: 300 <9>
    jwksMinRefreshPauseSeconds: 1 <10>
    enableECDSA: "true" <11>
----
<1> Listener type set to `oauth`.
<2> URI of the token issuer used for authentication.
<3> URI of the JWKS certificate endpoint used for local JWT validation.
<4> The token claim (or key) that contains the actual user name in the token. The user name is the _principal_ used to identify the user. The `userNameClaim` value will depend on the authentication flow and the authorization server used.

> <5> (Optional) Setting this option activates the re-authentication mechanism which enforces session expiry to that of the access token. If the specified value is less than the time left for the access token to expire, then the client will have to re-authenticate before the actual token expiry. By default, the session does not expire when access token expires, and the client attempts no re-authentication.

Reviewer: ```suggestion
<5> (Optional) Activates the Kafka re-authentication mechanism that enforces session expiry to the same length of time as the access token. If the specified value is less than the time left for the access token to expire, then the client will have to re-authenticate before the actual token expiry. By default, the session does not expire when the access token expires, and the client does not attempt re-authentication.
```

> <5> (Optional) Activates the Kafka re-authentication mechanism that enforces session expiry to the same length of time as the access token. If the specified value is less than the time left for the access token to expire, then the client will have to re-authenticate before the actual token expiry. By default, the session does not expire when the access token expires, and the client does not attempt re-authentication.

Reviewer: @mstruk - How many seconds should the user set for `maxSecondsWithoutReauthentication`? I think we should provide some guidance here?

Author: In the example it is set to 3600 seconds (1 hour) which is our guidance.

      secretName: my-cluster-oauth
      key: clientSecret
    userNameClaim: preferred_username <4>
    maxSecondsWithoutReauthentication: 3600 <5>
----
--
<1> URI of the token introspection endpoint.
<2> Client ID to identify the client.
<3> Client Secret and client ID is used for authentication.
<4> The token claim (or key) that contains the actual user name in the token. The user name is the _principal_ used to identify the user. The `userNameClaim` value will depend on the authorization server used.

> <5> (Optional) Setting this option activates the re-authentication mechanism which enforces session expiry to that of the access token. If the specified value is less than the time left for the access token to expire, then the client will have to re-authenticate before the actual token expiry. By default, the session does not expire when access token expires, and the client attempts no re-authentication.

Reviewer: _Please edit <5> to match line 75._
